src/ECDSASignatureChecker.sol

```diff
@@ -0,0 +1,170 @@
+    ECDSAStakeRegistry public immutable stakeRegistry;
+    IDelegationManager public immutable delegation;
+    /// @notice If true, check the staleness of the operator stakes and that its within the delegation withdrawalDelayBlocks window.
+    bool public staleStakesForbidden;
+
+    modifier onlyCoordinatorOwner() {
+        require(msg.sender == registryCoordinator.owner(), "BLSSignatureChecker.onlyCoordinatorOwner: caller is not the owner of the registryCoordinator");
+        _;
+    }
+
+    constructor(ECDSARegistryCoordinator _registryCoordinator) {
+        registryCoordinator = _registryCoordinator;
+        stakeRegistry = ECDSAStakeRegistry(address(_registryCoordinator.stakeRegistry()));
+        delegation = stakeRegistry.delegation();
+        
+        staleStakesForbidden = true;
+    }
+
+    /**
+     * RegistryCoordinator owner can either enforce or not that operator stakes are staler
+     * than the delegation.withdrawalDelayBlocks() window.
+     * @param value to toggle staleStakesForbidden
+     */
+    function setStaleStakesForbidden(bool value) external onlyCoordinatorOwner {
+        staleStakesForbidden = value;
+        emit StaleStakesForbiddenUpdate(value);
+    }
+
+    /**
+     * @notice This function is called by disperser when it has aggregated all the signatures of the operators
+     * that are part of the quorum for a particular taskNumber and is asserting them into onchain. The function
+     * checks that the claim for aggregated signatures are valid.
+     *
+     * The thesis of this procedure entails:
+     * - getting the aggregated pubkey of all registered nodes at the time of pre-commit by the
+     * disperser (represented by apk in the parameters),
+     * - subtracting the pubkeys of all the signers not in the quorum (nonSignerPubkeys) and storing 
+     * the output in apk to get aggregated pubkey of all operators that are part of quorum.
+     * - use this aggregated pubkey to verify the aggregated signature under BLS scheme.
+     * 
+     * @dev Before signature verification, the function verifies operator stake information.  This includes ensuring that the provided `referenceBlockNumber`
+     * is correct, i.e., ensure that the stake returned from the specified block number is recent enough and that the stake is either the most recent update
+     * for the total stake (of the operator) or latest before the referenceBlockNumber.
+     * @param msgHash is the hash being signed
+     * @param quorumNumbers is the bytes array of quorum numbers that are being signed for
+     * @param signerIds TODO: document
+     * @param signatures TODO: document
+     * @return quorumStakeTotals is the struct containing the total and signed stake for each quorum
+     * @return signatoryRecordHash is the hash of the signatory record, which is used for fraud proofs
+     */
+    function checkSignatures(
+        bytes32 msgHash, 
+        bytes calldata quorumNumbers,
+        bytes32[] memory signerIds,
```

Comment:
should we be calling `signerIds` `signerAddresses` instead? 
They are verified in
```
checkSignature_EIP1271(address signer, bytes32 digestHash, bytes memory signature)
```
and passed as the signer (an address)

Reply:
fine by me!